contracts/events/TopicEvent.sol

```diff
@@ -123,15 +106,13 @@ contract TopicEvent is ITopicEvent, Ownable {
 
         version = _version;
         owner = _owner;
-        name = _name;
-        resultNames = _resultNames;
+        eventName = _name;
 
+        eventResults[0] = "Invalid";
+        numOfResults++;
         for (uint i = 0; i < _resultNames.length; i++) {
             if (!_resultNames[i].isEmpty()) {
-                balances[i] = ResultBalance({
-                    totalBets: 0,
-                    totalVotes: 0
-                    });
+                eventResults[i + 1] = _resultNames[i];
```

Comment:
Do you need to check length of _resultName in the beginning, cuz eventResults[i+1] could go out of index range if input's length is bigger.

Reply:
Not necessary, the inputs through cli will break if it doesn't match the abi so `_resultNames` will always only have a max length of 10.